### DISCUSSION TITLE
How to debug failed tramp connection?

### DISCUSSION THREAD

**Discussion by ileixe** [joaotavora/eglot#1153] [Q&A] [answered]
Hi, 

I think it might be off-topic for eglot itself, but it've affected my eglot workflow a lot so want to ask for help.

When I tried to  launch 'eglot' in remote (via tramp) project, it failed to connect from time to time. This is related log below

In message buffer, I can see rust-analyzer could not discover workspace

```
Tramp: Opening connection *EGLOT (furiosa-runtime/(rust-mode)) stderr* for furiosa@furiosa-t2.office.furiosa.cloud using sshx...done
Tramp: Opening connection EGLOT (furiosa-runtime/(rust-mode))<1> for furiosa@furiosa-t2.office.furiosa.cloud using sshx...done
[eglot] Server reports (type=1): rust-analyzer failed to discover workspace [7 times]
```

In eglot log, I could find `rootUri` seems to be incorrect as it has "connecion failed" string 

```
[client-request] (id:1) Fri Jan 13 14:27:54 2023:
(:jsonrpc "2.0" :id 1 :method "initialize" :params
          (:processId nil :rootPath "/home/furiosa/Youseok/runtime/furiosa-runtime/" :rootUri "file:///sshx%3Afuriosa@furiosa-t2.office.furiosa.cloud%3Achannel%2021%3A%20open%20failed%3A%20connect%20failed%3A%20open%20failed%0D" :initializationOptions #s(hash-table size 1 test eql rehash-size 1.5 rehash-threshold 0.8125 data
```

So I imagine that as tramp could not connect correctly, it's clear that eglot could not launch correctly.

My question is that how to debug such connection failure? It's not consistently appearing and I don't have any other network issue (terminal workflow in the server via ssh works well).

Any advice would be appreciated.

**Comment by joaotavora** [comment, marked as answer]
Sorry for the long delay. If the problem still happens, please report this to Emacs's bug tracker and try to provide as clear a reproduction recipe as possible.  See https://debbugs.gnu.org/cgi/bugreport.cgi?bug=61350 for a good example.  CC me in the bug report email (my email address is in the source code).
